### PR TITLE
queryModifier often tries to sort by things that it shouldn't

### PR DESCRIPTION
**Pull request by pgebheim**
Story details: https://app.clubhouse.io/augur/story/10203

I've broken down sorting into two cases:

For fields which we auto-select as BN (using post-process-database-results.ts) -- we will sort in userland, and manually apply offset + limit;

For fields which are numerically sortable in the DB, we use a DB sort.

### CODE REVIEW

File: src/server/getters/database.ts [hunk 1]
```diff
@@ -25,13 +28,78 @@ export interface Dictionary {
...
+  let sortDescending: boolean = defaultSortOrder.toLowerCase() === "desc";
+
+  if (sortBy != null) {
+    sortField = sortBy;
+    if (typeof(isSortDescending) !== "undefined" && isSortDescending !== null) {
+      sortDescending = isSortDescending;
+    }
+  }
+
+  query.select(db.raw(`?? as "xMySorterFieldx"`, [sortField])).asCallback((error: Error|null, rows: Array<RowWithSort>) => {
```
Comment (epheph) [inline src/server/getters/database.ts]: double `??`

File: src/server/getters/database.ts [hunk 2]
```diff
@@ -25,13 +28,78 @@ export interface Dictionary {
...
+  callback: GenericCallback<Array<T>>,
+): void {
+  type RowWithSort = T & {xMySorterFieldx: BigNumber};
+
+  let sortField: string = defaultSortBy;
+  let sortDescending: boolean = defaultSortOrder.toLowerCase() === "desc";
+
+  if (sortBy != null) {
+    sortField = sortBy;
+    if (typeof(isSortDescending) !== "undefined" && isSortDescending !== null) {
```
Comment (epheph) [inline src/server/getters/database.ts]: what if they passed in `isSortDescending: false` ?

Comment (pgebheim) [reply src/server/getters/database.ts]: If so, the. The nullity and undefined checks will pass and sortDescending fill be set to false on line 65.

File: src/server/getters/database.ts (hunk 1 above)
Comment (epheph) [reply src/server/getters/database.ts]: ahh cool!

File: src/server/getters/database.ts (hunk 2 above)
Comment (epheph) [reply src/server/getters/database.ts]: oop, see it now!

**Pull request merged**